axe.d.ts

```diff
@@ -132,7 +132,7 @@ declare namespace axe {
   interface RunOptions {
     runOnly?: RunOnly | TagValue[] | string[] | string;
     rules?: RuleObject;
-    reporter?: ReporterVersion;
+    reporter?: ReporterVersion | string;
```

Comment:
I don't think this adds any value in typescript. The resolved type of this is just `string` and it looks like it overrides any value of the `ReporterVersion`. Might just make this `string` and not have the set values.

<img width="329" alt="screenshot showing that the resolved type of a set list of strings OR string is just string" src="https://github.com/dequelabs/axe-core/assets/2433219/d41e2bc0-27a6-4ad3-8340-6715b149aee6">

<img width="409" alt="screenshot of the resolved type when using a variable of type that is a set list of strings OR string is just string" src="https://github.com/dequelabs/axe-core/assets/2433219/b7cfaf42-d781-433f-a93e-c628934953ff">

Reply:
Sure, but can we do better? I looked at using a generic for this, something like:

```ts
interface RunOptions <T extends string | void = void>{
    reporter?: ReporterVersion | T;
}
```

That's fine, but then you can't pass this into axe.run because it expects `RunOptions<void>`. To fix that you'd have to make the reporter something that is passed into axe.run, and to AxeResults (for toolOptions), and AxeReporter, and getFrameContexts... It's a lot. I don't think that's worth the complexity. Open to suggestions though.

Reply:
Looking at a GitHub-wide code search for `"axe.addReporter"`, there are exactly zero folks who are actually using `axe.addReporter` in practice in an open source project (all hits are copies of axe-core or its tests).

Based on that, I definitely don't think making this more accurate is worth making the entire `RunOptions` type much more complex, and I don't think it's worth making the intellisense worse (like steve's screenshots point out) either; my inclination is to optimize for folks that aren't using `addReporter` and actually just do this:

```ts
  /** If you've added a custom ReporterVersion with `axe.addReporter`, that is also valid here */
  reporter?: ReporterVersion
```

In practice, I think if we're improving this type, the more important improvement would be to add the existing `AxeReporter` function as an allowable value.

Reply:
(I can also live with `ReporterVersion | string` if Wilco feels strongly about it - I only feel strongly that we shouldn't pollute the whole `RunOptions` type with a generic over this)

Reply:
I do prefer `ReporterVersion | string` here.